mls-rs/src/group/message_processor.rs

```diff
@@ -140,11 +140,60 @@ pub enum CommitEffect {
     NewEpoch(Box<NewEpoch>),
     Removed {
         new_epoch: Box<NewEpoch>,
-        remove_proposal: ProposalInfo<RemoveProposal>,
+        remover: Sender,
     },
     ReInit(ProposalInfo<ReInitProposal>),
 }
 
+impl MlsSize for CommitEffect {
+    fn mls_encoded_len(&self) -> usize {
+        0u8.mls_encoded_len()
+            + match self {
+                Self::NewEpoch(e) => e.mls_encoded_len(),
+                Self::Removed { new_epoch, remover } => {
+                    new_epoch.mls_encoded_len() + remover.mls_encoded_len()
+                }
+                Self::ReInit(r) => r.mls_encoded_len(),
+            }
+    }
+}
+
+impl MlsEncode for CommitEffect {
+    fn mls_encode(&self, writer: &mut Vec<u8>) -> Result<(), mls_rs_codec::Error> {
+        match self {
+            Self::NewEpoch(e) => {
+                1u8.mls_encode(writer)?;
+                e.mls_encode(writer)?;
+            }
+            Self::Removed { new_epoch, remover } => {
+                2u8.mls_encode(writer)?;
+                new_epoch.mls_encode(writer)?;
+                remover.mls_encode(writer)?;
+            }
+            Self::ReInit(r) => {
+                3u8.mls_encode(writer)?;
+                r.mls_encode(writer)?;
+            }
+        }
+
+        Ok(())
+    }
+}
+
+impl MlsDecode for CommitEffect {
+    fn mls_decode(reader: &mut &[u8]) -> Result<Self, mls_rs_codec::Error> {
+        match u8::mls_decode(reader)? {
+            1u8 => Ok(Self::NewEpoch(NewEpoch::mls_decode(reader)?.into())),
+            2u8 => Ok(Self::Removed {
+                new_epoch: NewEpoch::mls_decode(reader)?.into(),
+                remover: Sender::mls_decode(reader)?,
+            }),
+            3u8 => Ok(Self::ReInit(ProposalInfo::mls_decode(reader)?)),
+            _ => Err(mls_rs_codec::Error::UnsupportedEnumDiscriminant),
+        }
+    }
+}
+
```

Comment:
We do have at least some support for enum with associated data in the macro. Was there something in particular that was a problem?

Reply:
Mostly the problem was not supporting named fields --- `Removed { new_epoch }` --- I didn't want to blow up this PR by implementing that, I could create an issue

Reply:
Ok yeah lets make an issue for it, I think this can be solved

Reply:
#232